src/services/codefixes/helpers.ts

```diff
@@ -34,12 +32,22 @@ namespace ts.codefix {
         const name = declaration.name ? declaration.name.getText() : undefined;
         const visibility = getVisibilityPrefixWithSpace(getModifierFlags(declaration));
 
+        const typeAtNewDeclaration = checker.getTypeOfSymbolAtLocation(symbol, enclosingDeclaration);
+
         switch (declaration.kind) {
             case SyntaxKind.GetAccessor:
             case SyntaxKind.SetAccessor:
             case SyntaxKind.PropertySignature:
             case SyntaxKind.PropertyDeclaration:
-                const typeString = checker.typeToString(type, enclosingDeclaration, TypeFormatFlags.None);
+                let typeString: string | undefined = undefined;
+                const typeAtOldDeclaration = checker.getTypeAtLocation(declaration);
+                if ((typeAtOldDeclaration as TypeParameter).isThisType) {
+                    typeString = "this";
+                }
+                else {
+                    typeString = checker.typeToString(typeAtNewDeclaration, enclosingDeclaration, TypeFormatFlags.None);

```

Comment:
You could also consider another `TypeFormatFlags` member named `PreserveThisTypes`